### ISSUE TITLE
Bomb explosions are glitched/creating stock explosions

### ISSUE THREAD

**Issue by Sandrik2** [PapaJoesSoup/BDArmory#675]
KSP Version: 1.6.1

BDArmory Version: 1.2.4

Physics Range Extender Version: 1.9.0

Mods Currently Installed:

AirplanePlus
BDArmory
Firespitter
Kopernicus
PhysicsRangeExtender
scatterer
TextureReplacer

(ModuleManager 4.0.2 is installed)

![Capture](https://user-images.githubusercontent.com/49348712/55673635-8740c700-5878-11e9-8027-046f2d1eaacf.PNG)

System Specs:

GeForce GTX 1050 Ti (Driver Version 419.67)
Intel Core i5-8300h at 2.30GHz base, 3.9 GHz turbo
8GB of DDR4 RAM (7.85GB usable)
Running at 1920 x 1080, 60Hz

![Capture2](https://user-images.githubusercontent.com/49348712/55673659-de469c00-5878-11e9-9cb0-634e845130dd.PNG)

Issue:

All of BDArmory is working properly, and even the Mk82 and cluster bombs damage targets. My issue is that whenever I do drop a Mk82 or a cluster bomb, it just creates stock explosions, despite still damaging the target. I installed everything correctly, and I am dropping them with the weapons manager as usual. 

Test: 

I just used a stock Aeris 3A and attached the following to the wings

(2) Cluster Bombs
(2) Mk82 Bombs
(2) JDAM Bombs
(2) Mk82 SnakeEye Bombs
and I attached a all the different missiles as well. (I also attached a Weapon Manager).

All of these weapons DO NOT produce proper explosions when dropped.

They cause explosions that look like this:

![Kerbal Space Program Screenshot 2019 04 06 - 14 42 46 35](https://user-images.githubusercontent.com/49348712/55673780-78f3aa80-587a-11e9-8076-4cb3085c0730.png)

As opposed to this:

![Capture3](https://user-images.githubusercontent.com/49348712/55673797-bbb58280-587a-11e9-9665-2a523133b620.PNG)

Note: The "Detonate" button on bombs does not work at all. No explosion happens.

Extra Note: I also installed BDA and Physics Range Extender with the ModuleManager by themselves, but the issue persisted. 

KSP Logs:

[output_log.txt](https://github.com/PapaJoesSoup/BDArmory/files/3050822/output_log.txt)

Thank you!

**Comment by Gedas-S** [comment]
@Sandrik2 I don't see anything weird in the logs. I see detonations triggering, and if you say they even damage the parts, most of the code should be working.

My guess is that one of the graphics enhancement mods is eating the explosion texture. It is not very likely, as I know some of the team use those mods and haven't reported any issues, but it could be something in a new release.

Could you try if the issue persists without scatterer/texturereplacer?

**Comment by Sandrik2** [comment]
Yeah when I get on KSP later today I will check. Thanks for the quick response!

**Comment by Sandrik2** [comment]
Hey, just got back and got on KSP (was a long day) and I loaded up just BDArmory, PhysicsRangeExtender, and ModuleManager 4.0.2. Dropped the bombs as usual and the same issue is happening, stock like explosions. Tbh it doesn't bother me that much, but it is still frustrating to not have the complete explosion effects as they look very nice for the mod. 

Here's a screen again:
![Kerbal Space Program Screenshot 2019 04 08 - 21 57 33 22](https://user-images.githubusercontent.com/49348712/55768156-b3f80880-5a49-11e9-8181-d22bc4a459b6.png)

And the logs again if you need them:
[output_log.txt](https://github.com/PapaJoesSoup/BDArmory/files/3056871/output_log.txt)

Thank you once again!

**Comment by Sandrik2** [comment]
Um, hey. It's been 10 days and I would really want to just get some sort of response or fix to this issue, any help?

**Comment by Gedas-S** [comment]
@Sandrik2 sorry, I looked through the log, and I'm out of ideas. I'm running pretty much the same setup, and I cannot seem to replicate it on my box. So right now I don't really have any advice or resolution. :(

Thank you for the report!

**Comment by Sandrik2** [comment]
Alright well thanks for trying anyways. I can still play fine it just bugs me the explosions are so small. Thanks for the help again, and if you come upon a solution then please tell me.

**Comment by AcidBurn9** [comment]
#629 
Could be the same thing

**Comment by CamaradeKC** [comment]
i had the same problem i just increase physics range thx to physicsrangeextender and it fixed the problem. the physics juste don't applied on the bomb that were to far so the explosion cant happen

**Comment by sandroimerlishvili** [comment]
@jrodrigv Hello, I just wanted to put an update that I am still experiencing this same exact issue on KSP 1.12.2 with the latest versions of BDA+ and PRE. I can open a new issue, but I am running the same exact setup as described here and wanted to know if there were any updates on the problem prior to creating another post. Do you have any idea? It's as if the textures used by BDA are simply not in the game, and everything (ex. bombs and tank turrets) only make stock explosions and refuse to create the BDA explosion texture.

**Comment by REE-boop** [comment]
> @Sandrik2 sorry, I looked through the log, and I'm out of ideas. I'm running pretty much the same setup, and I cannot seem to replicate it on my box. So right now I don't really have any advice or resolution. :(
> 
> Thank you for the report!

are you saying you get the good explosions? i thought it was just a thing with the new bdarmory (plus) that im using, what version of ksp and bdarmory are you running?

**Comment by REE-boop** [comment]
> i had the same problem i just increase physics range thx to physicsrangeextender and it fixed the problem. the physics juste don't applied on the bomb that were to far so the explosion cant happen

what did you change it to and how?

**Comment by REE-boop** [comment]
> KSP Version: 1.6.1
> 
> BDArmory Version: 1.2.4
> 
> Physics Range Extender Version: 1.9.0
> 
> Mods Currently Installed:
> 
> AirplanePlus BDArmory Firespitter Kopernicus PhysicsRangeExtender scatterer TextureReplacer
> 
> (ModuleManager 4.0.2 is installed)
> 
> ![Capture](https://user-images.githubusercontent.com/49348712/55673635-8740c700-5878-11e9-8027-046f2d1eaacf.PNG)
> 
> System Environment:
> 
> OS: Windows 10 Home 64bit
> 
> System Specs:
> 
> GeForce GTX 1050 Ti (Driver Version 419.67) Intel Core i5-8300h at 2.30GHz base, 3.9 GHz turbo 8GB of DDR4 RAM (7.85GB usable) Running at 1920 x 1080, 60Hz
> 
> ![Capture2](https://user-images.githubusercontent.com/49348712/55673659-de469c00-5878-11e9-9cb0-634e845130dd.PNG)
> 
> Issue:
> 
> All of BDArmory is working properly, and even the Mk82 and cluster bombs damage targets. My issue is that whenever I do drop a Mk82 or a cluster bomb, it just creates stock explosions, despite still damaging the target. I installed everything correctly, and I am dropping them with the weapons manager as usual.
> 
> Test:
> 
> I just used a stock Aeris 3A and attached the following to the wings
> 
> (2) Cluster Bombs (2) Mk82 Bombs (2) JDAM Bombs (2) Mk82 SnakeEye Bombs and I attached a all the different missiles as well. (I also attached a Weapon Manager).
> 
> All of these weapons DO NOT produce proper explosions when dropped.
> 
> They cause explosions that look like this:
> 
> ![Kerbal Space Program Screenshot 2019 04 06 - 14 42 46 35](https://user-images.githubusercontent.com/49348712/55673780-78f3aa80-587a-11e9-8076-4cb3085c0730.png)
> 
> As opposed to this:
> 
> ![Capture3](https://user-images.githubusercontent.com/49348712/55673797-bbb58280-587a-11e9-9665-2a523133b620.PNG)
> 
> Note: The "Detonate" button on bombs does not work at all. No explosion happens.
> 
> Extra Note: I also installed BDA and Physics Range Extender with the ModuleManager by themselves, but the issue persisted.
> 
> KSP Logs:
> 
> [output_log.txt](https://github.com/PapaJoesSoup/BDArmory/files/3050822/output_log.txt)
> 
> Thank you!

hey dude i figured it out, you gotta go into your bdarmory folder >models then install the original bdarmory and drag and drop the explosion folder in the original bdarmory into the models folder of the bdarmory you are using, this should give you the original effects (they will be changed a bit but this is due the bombs having different code and this being a newer version i think)
